### PR TITLE
[Component/Pipeline]: Add Typescript/JavaScript samples for SDKv2

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description
 - add component Typescript/JavaScript samples
 - update dependencies in package.json, since file:../azure-arm-machinelearning-2.1.0/package doesn't exists in repository directory 

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/typescript/src/resources/component/componentVersionsListSample.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ *
+ * @summary List component versions.
+ * x-ms-original-file: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2022-10-01/examples/ComponentVersion/list.json
+ */
+async function listComponentVersion() {
+  const name = "command_component_basic";
+  const orderBy = "createdtime desc";
+  const top = 1;
+  const options: ComponentVersionsListOptionalParams = { orderBy, top };
+  const resArray = new Array();
```
Comment (guanghechen) [inline sdk/typescript/src/resources/component/componentVersionsListSample.ts:26]: Could you add type for the `resArray`? Something like `const resArray = new Array<Item>();`

File: sdk/typescript/src/resources/component/componentVersionsCreateOrUpdateSample.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+const resourceGroupName = getEnvironmentVariable("RESOURCEGROUP_NAME");
+const workspaceName = getEnvironmentVariable("WORKSPACE_NAME");
+
+/**
+ * This sample demonstrates how to Create or update version.
+ *
+ * @summary Create or update version.
+ * x-ms-original-file: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2022-10-01/examples/ComponentVersion/createOrUpdate.json
+ */
+async function createOrUpdateComponentVersion() {
```
Comment (guanghechen) [inline sdk/typescript/src/resources/component/componentVersionsCreateOrUpdateSample.ts:22]:
```suggestion
async function createOrUpdateComponentVersion(): Promise<void> {
```

File: sdk/typescript/src/resources/component/componentVersionsDeleteSample.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+const resourceGroupName = getEnvironmentVariable("RESOURCEGROUP_NAME");
+const workspaceName = getEnvironmentVariable("WORKSPACE_NAME");
+
+/**
+ * This sample demonstrates how to Delete version.
+ *
+ * @summary Delete version.
+ * x-ms-original-file: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2022-10-01/examples/ComponentVersion/delete.json
+ */
+async function deleteComponentVersion() {
```
Comment (guanghechen) [inline sdk/typescript/src/resources/component/componentVersionsDeleteSample.ts:19]:
```suggestion
async function deleteComponentVersion(): Promise<void> {
```

File: sdk/typescript/src/resources/component/componentVersionsListSample.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+const resourceGroupName = getEnvironmentVariable("RESOURCEGROUP_NAME");
+const workspaceName = getEnvironmentVariable("WORKSPACE_NAME");
+
+/**
+ * This sample demonstrates how to List component versions.
+ *
+ * @summary List component versions.
+ * x-ms-original-file: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2022-10-01/examples/ComponentVersion/list.json
+ */
+async function listComponentVersion() {
```
[the remark this answers is not in the document]
Comment (YingChen1996) [reply sdk/typescript/src/resources/component/componentVersionsListSample.ts:21]: Thanks, added

File: sdk/typescript/src/resources/component/componentVersionsListSample.ts (hunk 1 above)
Comment (YingChen1996) [reply sdk/typescript/src/resources/component/componentVersionsListSample.ts:26]: Thanks, updated

**Review by zhengfeiwang — APPROVED** [review]

File: sdk/typescript/package.json [hunk 5]
```diff
@@ -31,7 +31,7 @@
   },
   "homepage": "https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/machinelearning/arm-machinelearning",
   "dependencies": {
-    "@azure/arm-machinelearning": "file:../azure-arm-machinelearning-2.1.0/package",
+    "@azure/arm-machinelearning": "^2.1.0",
```
Comment (zhengfeiwang) [inline sdk/typescript/package.json:34]: Any background on this change?

Comment (YingChen1996) [reply sdk/typescript/package.json:34]: sure, will update in pr description.

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**